department-of-veterans-affairs/va.gov-cms
# VACMS-14290 implement max depth magichead

## Description

Adds a new 'max depth' option to magichead field form display settings. The max depth is passed to core's tableDrag API as the 'limit' property which allows us to leverage core for the primary behavior of preventing the maximum depth from being exceeded via drag-and-drop.

Relates to #14290 

## Testing done
Tested in the UI locally (verifying drag-and-drop behavior meets ACs)

## Screenshots
![Screenshot 2023-08-08 at 1 19 06 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/d8dfcbfb-19c1-4c14-8d96-00a05581ec33)

## QA steps
As a content admin
1. Visit /admin/structure/taxonomy/manage/va_benefits_taxonomy/overview/form-display
2. Find the 'Eligibility overview sections' field and click the Settings cog on the far right side of the table.
   - [ ]  In the expanded page, verify the 'Max Depth' option exist and has a value of '3'.
3. Visit /admin/structure/taxonomy/manage/va_benefits_taxonomy/add
4. Fill out test content for all required fields.
5. Create 5 'Overview sections' (in the Eligibility overview sections' fieldset)
6. Click 'Save and go to list'
7. On the list page, click 'Edit' next to the term you just created
8. Scroll to the Eligibility overview section.
9. Starting with the bottom element, drag it under the above item, creating a tree with single root parent
   - [ ] Validate you are not allowed to nest items with a depth greater than 3


### Definition of Done


### Select Team for PR review
